src/data/nat/digits.lean

```diff
@@ -90,7 +90,28 @@ begin
         rw nat.div_eq_of_lt w₂,
         dsimp only [digits_aux_zero],
         rw nat.mod_eq_of_lt w₂, } } }
-end.
+end
+
+/-- The digits in the base b+2 expansion of n are all less than b+2 -/
+lemma digits_lt_base' (b : ℕ) (m : ℕ) : ∀ d ∈ digits (b+2) m, d < b+2 :=
+begin
+  apply nat.strong_induction_on m,
+  intros n IH d hd,
+  unfold digits at hd IH,
+  cases n with n,
+  { cases hd}, -- base b+2 expansion of 0 has no digits
+  rw digits_aux_def (b+2) (by linarith) n.succ (nat.zero_lt_succ n) at hd,
+  cases hd,
+  { rw hd, exact n.succ.mod_lt (by linarith)},
```

Comment:
Just as a comment/question as I worked on a proof in parallel to Kevin, is there a significant difference between using `(by linarith)` versus `dec_trivial` here?

Reply:
`dec_trivial`s probably marginally faster. Not a big deal either way.